internal/source/git.go

```diff
@@ -98,6 +112,23 @@ func (r *Git) Unpack(ctx context.Context, bundle *rukpakv1alpha1.Bundle) (*Resul
 	return &Result{Bundle: bundleFS, ResolvedSource: resolvedSource, State: StateUnpacked}, nil
 }
 
+func (r *Git) getCredential(ctx context.Context, bundle *rukpakv1alpha1.Bundle) (string, string, error) {
```

Comment:
Thanks!  I updated the function header.  It made easy to read.

Reply:
Just for future reference, putting names in the function signatures return values has a few side effects:
1. When the function scope is entered, those names are instantiated as variables with the associated return type
2. It allows [naked returns](https://go.dev/tour/basics/7), which IMO are a _hugely_ bad practice because it makes it much harder to mentally understand what's being returned at the return site.

I basically never use named return values for this reason (and instead, I include GoDoc that describe the return values).

I'd honestly be in favor of a linter that outright blocked this kind of thing in the future.

Reply:
GoDoc is another option for sure. I'd be more comfortable with updating the comments with this to describe the return values based on what you're saying Joe. Unresolving.

Reply:
OK.  I put it back and adding the comment for the function.